tests/pypi/extension/extension_tests.bzl

```diff
@@ -407,6 +407,97 @@ new-package==0.0.1 --hash=sha256:deadb00f2
 
 _tests.append(_test_simple_multiple_python_versions)
 
+def _test_simple_multiple_extras_same_whl(env):
+    """Test that reproduces an issue where multiple extras point to same whl.
+
+    Based on https://github.com/bazel-contrib/rules_python/issues/2797#issuecomment-3143914644.
+    """
+    pypi = _parse_modules(
+        env,
+        module_ctx = _mock_mctx(
+            _mod(
+                name = "rules_python",
+                parse = [
+                    _parse(
+                        hub_name = "pypi",
+                        python_version = "3.12",
+                        download_only = True,
+                        requirements_by_platform = {
+                            "requirements.linux_arm64.txt": "linux_aarch64",
+                            "requirements.linux_x86_64.txt": "linux_x86_64",
+                        },
+                        experimental_index_url = "pypi.org",
+                    ),
+                ],
+            ),
+            read = lambda x: {
+                "requirements.linux_arm64.txt": """\
+package==0.7.0 \
+    --hash=sha256:4dd8924f171ed73a4f1a6191e2f800ae1745069989b69fabc45593d6b6504003 \
+    --hash=sha256:62833036cbaf4641d66ae94c61c0446890a91b2c0d153946583a0ebe04877a76
+""",
+                "requirements.linux_x86_64.txt": """\
+package[extra]==0.7.0 \
+    --hash=sha256:62833036cbaf4641d66ae94c61c0446890a91b2c0d153946583a0ebe04877a76
+""",
```

Comment:
I am testing this in `main` and it seems that a universal requirements file might work better in this particular case - I cannot create a failing test case for `parse_requirements` function yet.

Reply:
Actually, `parse_requirements` is not handling this correctly.

It returns output that will not work with the current implementation of the `pip.parse`, so it is to blame, but I need to look a little more at it where to do the fix.